Counterexamples/EulerSumOfPowers.lean

```diff
@@ -0,0 +1,387 @@
+/-
+Copyright (c) 2025 Snir Broshi. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Snir Broshi, Michael Stoll
+-/
+import Mathlib.NumberTheory.FLT.Three
+
+/-!
+# Euler's sum of powers conjecture
+
+Euler's sum of powers conjecture says that at least n nth powers of positive integers
+are required to sum to an nth power of a positive integer.
+
+This was an attempt to generalize Fermat's Last Theorem,
+which is the special case of summing 2 nth powers.
+
+We demonstrate the connection with FLT, prove that it's true for `n ≤ 3`,
+and provide counterexamples for `n = 4` and `n = 5`.
+The status of the conjecture for `n ≥ 6` is unknown.
+
+https://en.wikipedia.org/wiki/Euler%27s_sum_of_powers_conjecture
+http://euler.free.fr/
+-/
+
+
+namespace Counterexample
+
+/-- Euler's sum of powers conjecture over a given semiring with a specific exponent. -/
+abbrev SumOfPowersConjectureWith (R : Type*) [Semiring R] (n : ℕ) : Prop :=
+  ∀ (a : List R) (b : R), 2 ≤ a.length → 0 ∉ a → b ≠ 0 →
+  (a.map (· ^ n)).sum = b ^ n → n ≤ a.length
+
+/-- Euler's sum of powers conjecture over the naturals for a given exponent. -/
+abbrev SumOfPowersConjectureFor (n : ℕ) : Prop := SumOfPowersConjectureWith ℕ n
+
+/-- Euler's sum of powers conjecture over the naturals. -/
+abbrev SumOfPowersConjecture : Prop := ∀ n, SumOfPowersConjectureFor n
+
+/-- Euler's sum of powers conjecture over the integers for a given exponent. -/
+abbrev SumOfPowersConjectureIntFor (n : ℕ) : Prop := SumOfPowersConjectureWith ℤ n
+
+/-- Euler's sum of powers conjecture over the integers. -/
+abbrev SumOfPowersConjectureInt : Prop := ∀ n, SumOfPowersConjectureIntFor n
```

Comment:
I don't think these are really needed; they don't give much of an abbreviation, and any additional definition increases the cognitive load.

Reply:
Maybe the integer ones, but the naturals are the conjecture itself. The whole point of this file is to define the statement of the conjecture. FLT splits into the same 3 definitions I have: `FermatLastTheoremWith R n`, `FermatLastTheoremFor n`, `FermatLastTheorem`